File: Student-Attendance-Record-I.cpp
class Solution 
{
public:
    bool checkRecord(string s) 
    {
        int a_count = 0;
        int l_count = 0;
        for (int i = 0; i < s.length(); i++) 
        {
            if (s[i] == 'A') 
            {
                a_count++;
                l_count = 0;
                if (a_count >= 2) 
                {
                    return false;
                }
            } 
            else if (s[i] == 'L') 
            {
                l_count++;
                if (l_count >= 3) 
                {
                    return false;
                }
            } 
            else 
            {
                l_count = 0;
            }
        }
        return true;
    }
};

File: Student-Attendance-Record-I.java
class Solution {
    public boolean checkRecord(String s) {
        int aCount = 0;
        int lCount = 0;

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);

            if (ch == 'A') {
                aCount++;
                lCount = 0;
                if (aCount >= 2) {
                    return false;
                }
            } else if (ch == 'L') {
                lCount++;
                if (lCount >= 3) {
                    return false;
                }
            } else {
                lCount = 0;
            }
        }

        return true;
    }
}
